#164798866 implement signup

BukkyOmo opened the pull request:
#### Description
>This PR implements user signup with react.

#### Type of change
>Feature(Non-breaking change)

#### How should this be manually tested?
>Run npm run start:dev 
>Click on signup on the homepage

#### Checklist:
- [x] Create a signup component
- [x] Create a modal and form for signup
- [x] Create signup style
- [x] Write media queries

#### What are the relevant pivotal tracker stories?
[#164798866](https://www.pivotaltracker.com/n/projects/2320805/stories/164798866)

#### Screenshots (if appropriate)
Web view
<img width="1440" alt="Screenshot 2019-05-07 at 5 17 12 PM" src="https://user-images.githubusercontent.com/41617644/57316090-e9e1e880-70ec-11e9-8a1e-dcd212ea33a3.png">

Mobile view
<img width="1099" alt="Screenshot 2019-05-07 at 5 17 52 PM" src="https://user-images.githubusercontent.com/41617644/57316127-fd8d4f00-70ec-11e9-9309-867c45cdfd47.png">

BukkyOmo committed:
164798866: story(frontend) - implement signup

- [x] Create signup component

[Delivers #164798866]

BukkyOmo committed:
164798866: story(frontend) - implement signup

- [x] Create signup component

[Delivers #164798866]

Chiazokam committed:
164798870-task (frontend): enable social login

- enable social login
- confirm login success or failure
- write tests
- write end to end tests

[Delivers #164798870]

davidokonji committed:
165826407-story(story) Get a single article

 - add components for view single article
 - write the unit test for built components
 - write integration test for built components

[Delivers #165826407]

BukkyOmo committed:
164798866: story(frontend) - implement signup

- [x] Create a signup components
- [x] Create a modal and signup form in the modal
- [x] Style signup form
- [x] Write media queries

[Delivers #164798866]

BukkyOmo committed:
164798866: story(frontend) - implement signup

- [x] Create a signup components
- [x] Create a modal and signup form in the modal
- [x] Style signup form
- [x] Write media queries

[Delivers #164798866]

BukkyOmo committed:
164798866: story(frontend) - implement signup

- [x] Create a signup components
- [x] Create a modal and signup form in the modal
- [x] Style signup form
- [x] Write media queries

[Delivers #164798866]

BukkyOmo committed:
164798866: story(frontend) - implement signup

- [x] Create a signup components
- [x] Create a modal and signup form in the modal
- [x] Style signup form
- [x] Write media queries

[Delivers #164798866]

geekdanitek committed:
164798875-story(article): users can see articles

- Create components for listing of articles

[Finish #164798875]

paradisekelechi commented on cypress/integration/signup.spec.js:
```diff
@@ -0,0 +1,38 @@
+  beforeEach(() => {
+    cy.visit('/signup');
+  });
+  it('Image should have alt as signup-icon', () => {
```
@BukkyOmo Your test description should be readable and simple. For example, you could have 
```
it('should load an image element', () => {})
```
It is inside the test that you assert the attributes and other properties .

paradisekelechi commented on cypress/integration/signup.spec.js:
```diff
@@ -0,0 +1,38 @@
+      .should('have.attr', 'alt', 'signup-icon');
+  });
+    it('Button should have type submit', () => {
+    cy.get('img')
```
This is a repetition of the test above. More so, it should all come under a test case

paradisekelechi commented on cypress/integration/signup.spec.js:
```diff
@@ -0,0 +1,38 @@
+    cy.get('img')
+      .should('have.attr', 'alt', 'signup-icon');
+    });
+  it('Get the firstname input, type into it and verify that the value has been updated', () => {
```
@BukkyOmo 
```
it('should simulate the input action for the firstname', ()=> {})```

paradisekelechi commented on cypress/integration/signup.spec.js:
```diff
@@ -0,0 +1,38 @@
+describe('Test for Signup', () => {
```
The description should be simple
```
describe('Signup', () => {
```

paradisekelechi commented on package.json:
```diff
@@ -50,7 +50,7 @@
     "eslint-plugin-react": "^7.12.4",
     "express": "4.16.4",
-    "jest": "^24.7.1",
+    "jest": "^24.8.0",
```
Why the change to `24.8.0`?

paradisekelechi commented on src/__tests__/App.spec.js:
```diff
@@ -0,0 +1,10 @@
+   it('renders the various routes', () => {
+     expect(app.length).toEqual(1);
+    });
+});
```
There isn't any snapshot testing for this?

paradisekelechi commented on src/__tests__/actions/signupActions.spec.js:
```diff
@@ -0,0 +1,54 @@
+  password: 'johnny29383'
+};
+
+describe('Sign up Auth Actions', () => {
```
Good description. However, Signup should be one word.

paradisekelechi commented on src/actions/index.js:
```diff
@@ -1,3 +1,7 @@
+} from './authActions';
 
-export { increment, decrement };
+export {
```
Fix this hound issue

paradisekelechi commented on src/actions/index.js:
```diff
@@ -1,3 +1,7 @@
-export { increment, decrement };
+export {
+  signupBegin, signupSuccess, signupFailure, signUpUser
+};
```
Fix hound issues here

paradisekelechi commented on src/styles/components/_signupmodal.scss:
```diff
@@ -0,0 +1,86 @@
+  border-radius: 0;
+  height: 40px;
+  width: 22.5rem;
+}
```
Hound issues

BukkyOmo replied:
fixed

paradisekelechi commented on src/styles/components/_signupmodal.scss:
```diff
@@ -0,0 +1,86 @@
+.signup-form-button{
+  background-color: #40617B;
+  border-radius: 4px;
+  color:#fff;
```
These are legit hound issues. Fix them please

paradisekelechi commented:
Make sure you replace and fix the merge conflict and hound issues as quickly as possible.

chukwudidu committed:
165999824-bug(layout): fix bug with base layout

- include header and footer components in the routes index file
- fix white space bug in the header component
- fix bug vertical list in the footer for non-chrome browsers

paradisekelechi committed:
Merge pull request #28 from andela/bug-165999824-fix-for-layout-component

#165999824 Fix bug in base layout component

davidokonji committed:
166079475-bug(bug): Remove cypress build

- removed cypress build from circleci

[Delievers #166079475]

paradisekelechi committed:
Merge pull request #34 from andela/bug-166079475-remove-cypress

#166079475 Remove Cypress build from Circle CI

paradisekelechi committed:
Merge pull request #24 from andela/story-164798875-list-articles

#164798875 Users can see articles

paradisekelechi committed:
Merge pull request #26 from andela/story-165826407-view-article

#165826407 Users should be able to view a single article

paradisekelechi committed:
Merge pull request #32 from andela/story-164798870-social-login

#164798870 enable social login

BukkyOmo committed:
Merge branch 'story-164798866-implement-signup' of https://github.com/andela/forsetti-ah-frontend into story-164798866-implement-signup

BukkyOmo committed:
Merge branch 'story-164798866-implement-signup' of https://github.com/andela/forsetti-ah-frontend into story-164798866-implement-signup

BukkyOmo committed:
Merge branch 'story-164798866-implement-signup' of https://github.com/andela/forsetti-ah-frontend into story-164798866-implement-signup

BukkyOmo committed:
Merge branch 'story-164798866-implement-signup' of https://github.com/andela/forsetti-ah-frontend into story-164798866-implement-signup
